test/cppsuite/tests/search_near_03.cxx

```diff
@@ -0,0 +1,297 @@
+     * prefix already exists in the table. Otherwise if the record is not found, we can proceed to
+     * insert the full value.
+     * 4. Insert the full value (prefix, id).
+     * All of these operations are wrapped in the same transaction.
+     */
+    static bool
+    perform_unique_index_insertions(
+      thread_context *tc, scoped_cursor &cursor, collection &coll, std::string &prefix_key)
+    {
+        const char *key_tmp;
+        int exact_prefix, ret;
+
+        /* Insert the prefix. */
+        if (!tc->insert(cursor, coll.id, prefix_key))
+            return false;
+
+        /* Remove the prefix. */
+        if (!tc->remove(cursor, coll.id, prefix_key))
+            return false;
+
+        /*
+         * Prefix search near the prefix. We expect that the key is deleted and a WT_NOTFOUND error
+         * to be returned. If the key is present, it means the (prefix, id) has been inserted
+         * already, double check that the prefix portion match.
+         */
+        cursor->reconfigure(cursor.get(), "prefix_search=true");
+        cursor->set_key(cursor.get(), prefix_key.c_str());
+        ret = cursor->search_near(cursor.get(), &exact_prefix);
+        testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+        if (ret == 0) {
+            cursor->get_key(cursor.get(), &key_tmp);
+            testutil_assert(exact_prefix == 1);
+            testutil_assert(prefix_key == std::string(key_tmp).substr(0, prefix_key.size()));
+            return false;
+        }
+
+        /* Now insert the key with prefix and id. Use thread id to guarantee uniqueness. */
+        return tc->insert(cursor, coll.id, prefix_key + "," + std::to_string(tc->id));
+    }
+
+    static void
+    populate_worker(thread_context *tc)
+    {
+        logger::log_msg(LOG_INFO, "Populate with thread id: " + std::to_string(tc->id));
+
+        std::string prefix_key;
+        const uint64_t MAX_ROLLBACKS = 100;
+        uint32_t rollback_retries = 0;
+
+        /*
+         * Each populate thread perform unique index insertions on each collection, with a randomly
+         * generated prefix and thread id.
+         */
+        collection &coll = tc->db.get_collection(tc->id);
+        scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+        for (uint64_t count = 0; count < tc->key_count; ++count) {
+            tc->transaction.begin();
+            /*
+             * Generate the prefix key, and append a random generated key string based on the key
+             * size configuration.
+             */
+            prefix_key = random_generator::instance().generate_random_string(tc->key_size);
+            if (perform_unique_index_insertions(tc, cursor, coll, prefix_key)) {
+                tc->transaction.commit();
+            } else {
+                tc->transaction.rollback();
+                ++rollback_retries;
+                --count;
```

Comment:
I think this is a bad idea and we have the same issue in `search_near01`:
The first iteration count is set to 0, --count will make it to the highest `uint64_t` value which is not what we want. Check that `count` is not `0` before decrementing it.
What should we do about `search_near01`?

Reply:
I fixed it in search_near_01 one too.